src/token/api/src/canister.rs

```diff
@@ -22,25 +24,24 @@ use crate::types::{
 };
 
 use self::is20_transactions::{
-    batch_transfer, burn_as_owner, burn_own_tokens, claim, get_claim_subaccount, is20_transfer,
-    mint_as_owner, mint_test_token,
+    batch_transfer, burn_as_owner, burn_own_tokens, is20_transfer, mint_as_owner, mint_test_token,
 };
 
+#[cfg(feature = "claim")]
+use self::is20_transactions::{claim, get_claim_subaccount};
+
 mod inspect;
 
 pub mod icrc1_transfer;
 
+#[cfg(feature = "auction")]
 pub mod is20_auction;
 pub mod is20_transactions;
 
 pub(crate) const MAX_TRANSACTION_QUERY_LEN: usize = 1000;
 // 1 day in seconds.
 pub const DEFAULT_AUCTION_PERIOD_SECONDS: Timestamp = 60 * 60 * 24;
 
-pub fn pre_update<T: TokenCanisterAPI>(canister: &T, method_name: &str, method_type: MethodType) {
```

Comment:
Why this helper was removed? I added it for the sake of having pre updates being similar in the mocking canister and the original one. What if someone forgets to update a `pre_update` in one place or the other?

Reply:
The `pre_update` method is not a part of the general token API, but rather just a way we implement it for our default implementation of IS20 token. That's why it's correct place is in the `impl` crate and not here. This function is very different in other canisters that use IS20 API, for example in AMM, Neuro etc.